flow-model-generator/modules/flow-model-generator-ls-extension/src/main/java/io/ballerina/flowmodelgenerator/extension/ExpressionEditorService.java

```diff
@@ -61,6 +69,66 @@ public Class<?> getRemoteInterface() {
         return null;
     }
 
+    @JsonRequest
+    public CompletableFuture<VisibleVariableTypesResponse> visibleVariableTypes(VisibleVariableTypeRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            VisibleVariableTypesResponse response = new VisibleVariableTypesResponse();
+            try {
+                Path filePath = Path.of(request.filePath());
+                this.workspaceManager.loadProject(filePath);
+                Optional<SemanticModel> semanticModel = this.workspaceManager.semanticModel(filePath);
+                Optional<Document> document = this.workspaceManager.document(filePath);
+                if (semanticModel.isEmpty() || document.isEmpty()) {
+                    return response;
+                }
+
+                List<Category.Variable> moduleVariables = new ArrayList<>();
+                List<Category.Variable> configurableVariables = new ArrayList<>();
+                List<Category.Variable> localVariables = new ArrayList<>();
+                List<String> variableNames = new ArrayList<>();
+
+                semanticModel.get().moduleSymbols().stream()
+                        .filter(symbol -> symbol.kind() == SymbolKind.VARIABLE)
+                        .forEach(symbol -> {
+                            VariableSymbol variableSymbol = (VariableSymbol) symbol;
+                            String name = symbol.getName().orElse("");
+                            Type type = Type.fromSemanticSymbol(symbol);
+
+                            if (variableSymbol.qualifiers().contains(Qualifier.CONFIGURABLE)) {
+                                configurableVariables.add(new Category.Variable(name, type));
+
+                            } else {
+                                moduleVariables.add(new Category.Variable(name, type));
+                            }
+                            variableNames.add(name);
+                        });
+
+                semanticModel.get().visibleSymbols(document.get(), request.position()).stream()
+                        .filter(symbol -> symbol.kind() == SymbolKind.VARIABLE)
+                        .forEach(symbol -> {
+                            String name = symbol.getName().orElse("");
+                            Type type = Type.fromSemanticSymbol(symbol);
+
+                            if (!variableNames.contains(name)) {
```

Comment:
This condition doesn't guarantee we already have the correct variable name and type. Lets say we have `int a` at module level and `string a` at the local scope. This logic won't add the `string a` variable to the local variable list.